flink-metrics/flink-metrics-influxdb/src/main/java/org/apache/flink/metrics/influxdb/MetricMapper.java

```diff
@@ -71,8 +75,24 @@ static Point map(MeasurementInfo info, Instant timestamp, Meter meter) {
     }
 
     private static Point.Builder builder(MeasurementInfo info, Instant timestamp) {
-        return Point.measurement(info.getName())
-                .tag(info.getTags())
+        return Point.measurement(normalize(info.getName()))
+                .tag(normalize(info.getTags()))
                 .time(timestamp.toEpochMilli(), TimeUnit.MILLISECONDS);
     }
+
+    private static String normalize(String value) {
+        return value != null ? value.replace(POINT_DELIMITER, "") : null;
+    }
+
+    private static Map<String, String> normalize(Map<String, String> map) {
+        if (map != null) {
+            Map<String, String> normalizedMap = new HashMap<>(map.size() * 4 / 3 + 1);
+            for (Map.Entry<String, String> entry : map.entrySet()) {
+                normalizedMap.put(normalize(entry.getKey()), normalize(entry.getValue()));
+            }
+            return normalizedMap;
+        } else {
+            return null;
+        }
+    }
```

Comment:
The `MeasurementInfoProvider` already constructs a copy of the original map; normalize stuff there instead.

Reply:
ok